cfg/sorted_vector_helpers.cc

```diff
@@ -0,0 +1,59 @@
+#include "cfg/sorted_vector_helpers.h"
+
+using namespace std;
+
+namespace sorbet::cfg {
+void setDifferenceInplace(vector<int> &data, const vector<int> &toRemove) {
+    auto dataIt = data.begin();
+    auto removeIt = toRemove.begin();
+    while (dataIt != data.end() && removeIt != toRemove.end()) {
+        const int datum = *dataIt;
+        const int remove = *removeIt;
+        if (datum == remove) {
+            dataIt = data.erase(dataIt);
+        } else if (datum < remove) {
+            dataIt++;
+        } else {
+            removeIt++;
+        }
+    }
+}
+
+void setUnionInplace(vector<int> &into, const vector<int> &from) {
+    // A decent proportion of unions are just copies.
+    if (into.empty()) {
+        into = from;
+        return;
+    }
+
+    auto intoIt = into.begin();
+    auto fromIt = from.begin();
+
+    while (intoIt != into.end() && fromIt != from.end()) {
+        const auto intoEl = *intoIt;
+        const auto fromEl = *fromIt;
+        if (intoEl == fromEl) {
+            // Present in both sets.
+            fromIt++;
+            intoIt++;
+        } else if (intoEl < fromEl) {
+            // Present in into, not in from
+            intoIt++;
+        } else {
+            // intoEl > fromEl
+            // Present in from, not in into.
+            // Rather than insert in the middle of the vector, break out of loop and do a set union.
```

Comment:
This saves a small amount of time when unioning larger vectors. It's meager gains, so I can be convinced to use `insert` instead and have simpler code.

Reply:
This seems fine to me, too bad you can't use `absl::c_set_union` here :(